src/f77_interface.F

```diff
@@ -18,127 +18,131 @@
+                                      enumeration_type
+   USE input_keyword_types, ONLY: keyword_get, &
+                                  keyword_type
+   USE input_section_types, ONLY: &
+      section_get_keyword, section_release, section_type, section_vals_duplicate, &
+      section_vals_get, section_vals_get_subs_vals, section_vals_release, &
+      section_vals_remove_values, section_vals_retain, section_vals_type, section_vals_val_get, &
+      section_vals_write
+   USE kinds, ONLY: default_path_length, &
+                    default_string_length, &
+                    dp
+   USE machine, ONLY: default_output_unit, &
+                      m_chdir, &
+                      m_getcwd, &
+                      m_memory
+   USE message_passing, ONLY: mp_comm_type, &
+                              mp_comm_world, &
+                              mp_para_env_release, &
+                              mp_para_env_type, &
+                              mp_world_finalize, &
+                              mp_world_init
+   USE metadynamics_types, ONLY: meta_env_type
+   USE metadynamics_utils, ONLY: metadyn_read
+   USE mixed_environment_types, ONLY: mixed_environment_type
+   USE mixed_main, ONLY: mixed_create_force_env
+   USE mp_perf_env, ONLY: add_mp_perf_env, &
+                          get_mp_perf_env, &
+                          mp_perf_env_release, &
+                          mp_perf_env_retain, &
+                          mp_perf_env_type, &
+                          rm_mp_perf_env
+   USE nnp_environment, ONLY: nnp_init
+   USE nnp_environment_types, ONLY: nnp_type
+   USE offload_api, ONLY: offload_get_chosen_device, &
+                          offload_get_device_count, &
+                          offload_init, &
+                          offload_set_chosen_device
+   USE periodic_table, ONLY: init_periodic_table
+   USE pw_gpu, ONLY: pw_gpu_finalize, &
+                     pw_gpu_init
+   USE pwdft_environment, ONLY: pwdft_init
+   USE pwdft_environment_types, ONLY: pwdft_env_create, &
+                                      pwdft_environment_type
+   USE qmmm_create, ONLY: qmmm_env_create
+   USE qmmm_types, ONLY: qmmm_env_type
+   USE qmmmx_create, ONLY: qmmmx_env_create
+   USE qmmmx_types, ONLY: qmmmx_env_type
+   USE qs_environment, ONLY: qs_init
+   USE qs_environment_types, ONLY: get_qs_env, &
+                                   qs_env_create, &
+                                   qs_environment_type
+   USE reference_manager, ONLY: remove_all_references
+   USE string_table, ONLY: string_table_allocate, &
+                           string_table_deallocate
+   USE timings, ONLY: add_timer_env, &
+                      get_timer_env, &
+                      rm_timer_env, &
+                      timer_env_release, &
+                      timer_env_retain, &
+                      timings_register_hooks
+   USE timings_types, ONLY: timer_env_type
+   USE virial_types, ONLY: virial_type
+#if defined(__DLAF)
```

Comment:
Here the `#if defined(__DLAF)` could also be removed by makeing the `cp_dlaf_initialize/finalize` NoOps instead.

Reply:
The reason why I haven't done this is that I don't even include `fm/cp_dlfa_utils.c` nor `fm/cp_fm_dlaf_api.F` in the list of source files, if DLA-Future is not required.

Reply:
I had it the other way around, but changed it not to have empty functions referenced in the library. I can change it back if it's the preferred approach in CP2K.

Reply:
Yes, build flags should be used as little as possible because preprocessor statements always make for hard to read code.

Ideally there is only a single interface module for each flag and the rest of the code uses normal control flow.

In reality we rarely achieve this, but I'd appreciate if you could remove `__DLAF` from the high-level files: `f77_interface.F`, `environment.F`, `cp2k_runs.F`,  and `input_cp2k_global.F`.

Reply:
Done. Bonus: some of the problems related to https://github.com/cp2k/cp2k/issues/3071#issuecomment-1785370741 are gone.